apps/examples/smartfs_powercut/smartfs_powercut_main.c

```diff
@@ -498,30 +655,56 @@ int smartfs_powercut_main(int argc, char *argv[])
 #endif
 {
 	int ret;
-	printf("Init test to create test files\n");
+	struct stat st;
+	bool test_txt = false;
+
+	printf("Initializing test files\n");
 	ret = init_test_file();
 
 	if (ret != OK) {
-		printf("init_test_file failed\n");
+		printf("init_test_file() failed\n");
 		return ERROR;
 	}
-	printf("################### Init Finished, Test will started after 3sec ###################\n");
+
+	if (stat(TEST_SMALL_T, &st) == OK) {
+		test_txt = true;
+	}
+
+	printf("################### Init Finished, Test will start after 3sec ###################\n");
 	sleep(3);
 	while (1) {
 		/* Small test use same as size of data area of mtd sector, so there should not be error
 		 * during the verify for small file test because it protected by journal.
 		 */
-		ret = do_test(TEST_SMALL, TEST_SMALL_BACKUP, TEST_SMALL_BACKUP_DIR, TEST_SMALL_BUFSIZE, FILE_TYPE_RANDOM);
+		printf("Do test on %s\n", TEST_SMALL_R);
+		ret = do_test(TEST_SMALL_R, TEST_SMALL_R_BACKUP, TEST_SMALL_R_BACKUP_DIR, TEST_SMALL_BUFSIZE, FILE_TYPE_RANDOM);
 		if (ret != OK) {
-			printf("test small file failed\n");
+			printf("Powercut test on small random file failed\n");
 			return ret;
 		}
 
-		ret = do_test(TEST_LARGE, TEST_LARGE_BACKUP, TEST_LARGE_BACKUP_DIR, TEST_LARGE_BUFSIZE, FILE_TYPE_RANDOM);
+		printf("Do test on %s\n", TEST_LARGE_R);
+		ret = do_test(TEST_LARGE_R, TEST_LARGE_R_BACKUP, TEST_LARGE_R_BACKUP_DIR, TEST_LARGE_BUFSIZE, FILE_TYPE_RANDOM);
 		if (ret != OK) {
-			printf("test large file failed\n");
+			printf("Powercut test on large random file failed\n");
 			return ret;
 		}
+
+		if (test_txt) {
```

Comment:
This flag indicates if sample_file.txt was given by user
Which means we will now test text files also.

similar to reply on the other related comment

Reply:
you can create sample_file.txt in init_test_file. we don't need pre-load file by nxfuse or romfs.
If you want to test only one of this for short test, KConfig is correct. not local variable.
Because of restricted flash, there are many chances that we have to reduce size of contents for another model.
So we'd better create contents file after we realize bufsize & testcount.

Reply:
The size of sample text file does not depend on TEST_WRITECOUNT
We only need one full buffer worth of data
The same part of file will be copied again and again TEST_WRITECOUNT number of times to create main testing_ file and backup file.

so we only need sample_file.txt of size 8192 bytes

Reply:
Please refer to comment in line 517, buffer size has more chance that reduce its size than TEST_WRITECOUNT.
Because our target board always has very small ram size.
(Mediatek board, we have around hundreds Kilo bytes, without any service app now.)
So large size can be defined in KConfig, I think it will be better. But we don't have to do in this time.
Anyway, where is your source data? what is your plan?

Reply:
Referring to the same
Creating sample_file.txt during init()